### PR TITLE
Close files properly in `save_util.py`.

### PR DESCRIPTION
**Pull request by rhaps0dy**
Mark slow test with `expensive` (which is a defined marker in pyproject.tml) instead.

### CODE REVIEW

File: stable_baselines3/common/save_util.py [hunk 1]
```diff
@@ -308,28 +308,28 @@ def save_to_zip_file(
     :param pytorch_variables: Other PyTorch variables expected to contain name and value of the variable.
     :param verbose: Verbosity level: 0 for no output, 1 for info messages, 2 for debug messages
     """
-    save_path = open_path(save_path, "w", verbose=0, suffix="zip")
```
Comment (rhaps0dy) [inline stable_baselines3/common/save_util.py:311]: The diff here is terrible, I only added a `with` to this line and indented the rest.

File: stable_baselines3/common/save_util.py [hunk 2]
```diff
@@ -391,63 +391,62 @@ def load_from_zip_file(
     :return: Class parameters, model state_dicts (aka "params", dict of state_dict)
         and dict of pytorch variables
     """
-    load_path = open_path(load_path, "r", verbose=verbose, suffix="zip")
+    with open_path(load_path, "r", verbose=verbose, suffix="zip") as load_path:
```
Comment (rhaps0dy) [inline stable_baselines3/common/save_util.py:394]: Same here -- the only real difference is this line + indentations.

File: stable_baselines3/common/save_util.py (hunk 1 above)
Comment (dan-pandori) [reply stable_baselines3/common/save_util.py:311]: Ack, thank you for the heads up.

**Review by dan-pandori — APPROVED** [review]

**Comment by dan-pandori** [comment]
BTW, I think you merged this before my review. Which is I guess fine for something trivial like this in your own repo, but feels weird that you asked for review in that case :)

**Comment by rhaps0dy** [comment]
That's strange, it's not allowed to merge before review.

I did hit "auto-merge" before your review, but that makes it so when you hit the "approve" button it's instantly merged. Maybe I messed that up?

**Comment by rhaps0dy** [comment]
Screenshot 2023-10-13 at 14 35 09
Clearly I messed that up somehow

**Pull request merged**